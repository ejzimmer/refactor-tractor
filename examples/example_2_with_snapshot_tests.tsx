import React from 'react'
import { formatShortDate } from './example_1'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import dayjs from 'dayjs'


function DayOfTheWeekWidget({ date = '' }: { date?: string | number }) {
  if (!date) {
    return <div>No date entered</div>
  }

  const formattedDate = formatShortDate(date)
  const dayOfTheWeek = dayjs(date).format('dddd')

  return <div>{formattedDate} is a {dayOfTheWeek}</div>
}

describe('DayOfTheWeekWidget', () => {
  it('renders ?? when passed an empty string', () => {
    const { container } = render(<DayOfTheWeekWidget date="" />)
    expect(container).toMatchSnapshot()
  })

  it('renders ?? when passed the number 0', () => {
    const { container } = render(<DayOfTheWeekWidget date={0} />)
    expect(container).toMatchSnapshot()
  })

  it('renders ?? when passed a Tuesday', () => {
    const { container } = render(<DayOfTheWeekWidget date="2021-08-03" />)
    expect(container).toMatchSnapshot()
  })
})