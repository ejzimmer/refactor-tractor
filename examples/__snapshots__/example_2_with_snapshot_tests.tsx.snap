// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DayOfTheWeekWidget renders ?? when passed a Tuesday 1`] = `
<div>
  <div>
    2021-08-3
     is a 
    Tuesday
  </div>
</div>
`;

exports[`DayOfTheWeekWidget renders ?? when passed an empty string 1`] = `
<div>
  <div>
    No date entered
  </div>
</div>
`;

exports[`DayOfTheWeekWidget renders ?? when passed the number 0 1`] = `
<div>
  <div>
    No date entered
  </div>
</div>
`;
